spec/models/organization_spec.rb

```diff
@@ -8,15 +8,15 @@
       subject { organization }
 
       it { is_expected.to have_many(:api_secrets).through(:users) }
-      it { is_expected.to have_many(:articles) }
-      it { is_expected.to have_many(:listings) }
-      it { is_expected.to have_many(:collections) }
-      it { is_expected.to have_many(:credits) }
-      it { is_expected.to have_many(:display_ads) }
-      it { is_expected.to have_many(:notifications) }
+      it { is_expected.to have_many(:articles).dependent(:nullify) }
+      it { is_expected.to have_many(:collections).dependent(:nullify) }
+      it { is_expected.to have_many(:credits).dependent(:restrict_with_error) }
```

Comment:
Any reason this is not `:destroy`? Any reason `:restrict_with_error` handle is used? From what I read about it, it adds an error to the dependent object, but I'm not really understanding why it's only needed for credits.

Reply:
The logic here is: credits were purchased, so we want to avoid removing them automatically. The operator should be notified that credits exist associated to such an org and act accordingly. `:restrict_with_error` is helpful to build a UI on top of it so that you can tell the human what's going on.